src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IDiscardOperation.cs

```diff
@@ -0,0 +1,335 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    public partial class IOperationTests : SemanticModelTestBase
+    {
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void DiscardFlow_01()
+        {
+            string source = @"
+class C
+{
+    static void M()
+    /*<bind>*/{
+        _ = M2();
+    }/*</bind>*/
+
+    static int M2()
+    {
+        return 2;
+    }
+}
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            var expectedFlowGraph = @"
+Block[B0] - Entry
+    Statements (0)
+    Next (Regular) Block[B1]
+Block[B1] - Block
+    Predecessors: [B0]
+    Statements (1)
+        IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null) (Syntax: '_ = M2();')
+          Expression: 
+            ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Int32) (Syntax: '_ = M2()')
+              Left: 
+                IDiscardOperation (Symbol: System.Int32 _) (OperationKind.Discard, Type: System.Int32) (Syntax: '_')
+              Right: 
+                IInvocationOperation (System.Int32 C.M2()) (OperationKind.Invocation, Type: System.Int32) (Syntax: 'M2()')
+                  Instance Receiver: 
+                    null
+                  Arguments(0)
+
+    Next (Regular) Block[B2]
+Block[B2] - Exit
+    Predecessors: [B1]
+    Statements (0)
+";
+            VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>(source, expectedFlowGraph, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void DiscardFlow_02()
+        {
+            string source = @"
+class C
+{
+    static void M(int i)
+    /*<bind>*/{
+        i = M2(out var _);
```

Comment:
>i = M2(out var _); [](start = 8, length = 18)

For an ```out var``` case, it would be good to have a test with a control flow in an argument following the  ```out var _```. The test should verify that the discard is not spilled.

Reply:
@AlekseyTs , I've added the two tests you mentioned.  Can you take a look and if it's OK I'll merge.  Thanks.